### PR TITLE
SignalR Get Started - v7 prep

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review - versions 7, 6 and 3-5](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/signalr?view=aspnetcore-7.0&branch=pr-en-us-28590&tabs=visual-studio)

Prep for #28460 SignalR with JavaScript: Update to 7.0 VS - VSC - VS Mac

- Move SignalR Getting Started tutorial to new versioning by file system using includes, set all code and art paths accordingly.
- Some clean up where numbered steps jumped from numbers to bullet points towards the end on version 6.

### CODE REVIEW

**Comment by wadepickett** [comment]
@tdykstra, just a prep to version by file before doing a 7.0 update, but I cleaned it up a little too, so hence the review request.

File: aspnetcore/tutorials/signalr/includes/signalr3-5.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+:::moniker range=">= aspnetcore-3.1 < aspnetcore-6.0"
```
Comment (tdykstra) [inline aspnetcore/tutorials/signalr/includes/signalr3-5.md:1]: I believe we are supposed to put all prior versions in one include file.

File: aspnetcore/tutorials/signalr.md [hunk 2]
```diff
@@ -212,52 +213,53 @@ The preceding highlighted code adds SignalR to the ASP.NET Core dependency injec
...
-  ```dotnetcli
-  dotnet watch run --project SignalRChat.csproj
-  ```
+```dotnetcli
+dotnet watch run --project SignalRChat.csproj
+```
 
 # [Visual Studio for Mac](#tab/visual-studio-mac)
 
 1. From the menu, select **Run > Start Without Debugging**.
```
Comment (tdykstra) [inline aspnetcore/tutorials/signalr.md]: Looks like you missed a numbered step

File: aspnetcore/tutorials/signalr/includes/signalr6.md [hunk 3]
```diff
@@ -196,52 +197,53 @@ The preceding highlighted code adds SignalR to the ASP.NET Core dependency injec
...
-  ```dotnetcli
-  dotnet watch run --project SignalRChat.csproj
-  ```
+```dotnetcli
+dotnet watch run --project SignalRChat.csproj
+```
 
 # [Visual Studio for Mac](#tab/visual-studio-mac)
 
 1. From the menu, select **Run > Start Without Debugging**.
```
Comment (tdykstra) [inline aspnetcore/tutorials/signalr/includes/signalr6.md]: Did you leave this number here on purpose?

**Review by tdykstra — APPROVED** [review]
LGTM aside from a couple of minor comments.

What's the motivation for removing numbers from steps?

File: aspnetcore/tutorials/signalr/includes/signalr3-5.md (hunk 1 above)
Comment (wadepickett) [reply aspnetcore/tutorials/signalr/includes/signalr3-5.md:1]: Thanks Tom.  Looking at the example provided at the kickoff of file versioning, the organization was this:
https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/configuration

- Latest version (7.0 in this case)
  - include/<6.0 versioned file>
  - include/<3-5 version file>

So everything below 5 is grouped, all other older versions get their own file in an include?
Did the plan change maybe and I'm not up on the latest or am I maybe misunderstanding the plan?

File: aspnetcore/tutorials/signalr.md (hunk 2 above)
Comment (wadepickett) [reply aspnetcore/tutorials/signalr.md]: Hmm, sorry I missed that.  I remember doing a search to make sure I didn't miss one, and yet ...I still missed it.  Thanks.

File: aspnetcore/tutorials/signalr/includes/signalr6.md (hunk 3 above)
Comment (wadepickett) [reply aspnetcore/tutorials/signalr/includes/signalr6.md]: I just missed it somehow, will correct, thanks Tom for spotting it.

**Comment by wadepickett** [comment]
> LGTM aside from a couple of minor comments.
> 
> What's the motivation for removing numbers from steps?

- Rick preferred I don't do numbered steps.  So I have been avoiding them.
- I have come to appreciate why they don't always work well:
   -  For tabbed tutorials like this one, it is problematic to used numbered steps.  For example: Let's say there are 10 steps total in a section, and twice in those set of steps there are tabs for 3 different dev environment/IDE, but the number of steps for the tabbed sections differ.  One has 2 steps for VS and another 3 steps for VSC or 4 steps for VS for Mac.  The numbered steps end up being off, and then you see work arounds, like seemed to be done in this doc originally, where it jumps back and forth between numbers and bullets to make it all work out after the tabs.  
     - We also use aspnetcore/includes a lot in this repo.  So when those includes have steps, the the formatting of numbers/vs bullets also don't match. 
     - I personally like numbered steps in tutorials since it is easier to see steps one needs to take sorted out of other background information that is being provided, plus a quick reference to which step you left off on or had trouble on, and for consistency since the style guide recommends them, except in certain cases.  However, per points above, they don't always fit well with tab variations on the procedural content and/or includes.

File: aspnetcore/tutorials/signalr.md (hunk 2 above)
Comment (wadepickett) [reply aspnetcore/tutorials/signalr.md]: ..interesting.  I actually did have that numbered step fixed already before in commit [28590](https://github.com/dotnet/AspNetCore.Docs/pull/28590/commits/5118771af2c07cf6315b9739bb76ca04c1cbfe70) before I put in the review request.  I was was double checking all that, so I felt ashamed that I might have missed it anyway.  I think what happened is you hit it right away (thanks) and the build was generating, (which I should have waited to verify it was fine in the build before asking for a review, more shame, haha).

**Comment by Rick-Anderson** [comment]
> Update this topic and app samples to .NET 7.0

Why does this need a .NET 7 version, what's changed? Why not update to .NET 8 which is LTS?

If there are code changes, then the prep PR should include a copy of the old code so the next PR has only the differences.

File: aspnetcore/tutorials/signalr/includes/signalr6.md (hunk 3 above)
Comment (wadepickett) [reply aspnetcore/tutorials/signalr/includes/signalr6.md]: I guess that was already fixed in commit 5118771 before the review request, but wasn't showing on your end yet.  Already removed.

**Comment by wadepickett** [comment]
> > Update this topic and app samples to .NET 7.0
> 
> Why does this need a .NET 7 version, what's changed? Why not update to .NET 8 which is LTS?
> 
> If there are code changes, then the prep PR should include a copy of the old code so the next PR has only the differences.

The target version and sample would change at least to 7.  I need to look over what changed and test, which is part of the work/time reserved for this issue.  Then it is off to 8.0.

**Comment by wadepickett** [comment]
Verified in doc build that the code is all showing properly.

**Comment by Rick-Anderson** [comment]
> The target version and sample would change at least to 7. I need to look over what changed and test, which is part of the work/time reserved for this issue. Then it is off to 8.0. This is just the prep to organize before I do version 7.

Skip 7 and go straight to 8

File: aspnetcore/tutorials/signalr/includes/signalr3-5.md (hunk 1 above)
Comment (tdykstra) [reply aspnetcore/tutorials/signalr/includes/signalr3-5.md:1]: @Rick-Anderson Should we put all prior versions in one include?

**Pull request merged**